transport/transport_test.go

```diff
@@ -301,6 +298,89 @@ func setUpWithNoPingServer(t *testing.T, copts ConnectOptions, done chan net.Con
 	return tr
 }
 
+// TestMaxConnectionIdle tests that a server will send GoAway to a idle client.
+// An idle client is one who doesn't make any RPC calls for a duration of
+// MaxConnectionIdle time.
+func TestMaxConnectionIdle(t *testing.T) {
```

Comment:
It would be useful to have negative tests for this one and the third test -- i.e. make a call from the client every second for several seconds to ensure that the connection is not closed prematurely, and make sure the connection is not closed if the ping is responded to.